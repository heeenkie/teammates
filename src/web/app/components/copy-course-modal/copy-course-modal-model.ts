/**
 * Result of {@link CopyCourseModalComponent}
 */
export interface CopyCourseModalResult {
    newCourseId: string;
}