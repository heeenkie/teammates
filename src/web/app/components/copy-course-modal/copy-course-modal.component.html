<div class="modal-header">
    <h5 class="modal-title">
        <div>Copy this course with sessions</div>
    </h5>
    <button type="button" class="close" (click)="activeModal.dismiss()">
        <i class="fas fa-times"></i>
    </button>
</div>
<div id="copy-course-modal" class="modal-body">
    <div class="row">
        <div class="col-12">
            <div class="form-group">
                <label><b>Id for copied course</b></label>
                <input id="copy-course-id" type="text" class="form-control" [(ngModel)]="newCourseId" [maxlength]="COURSE_ID_MAX_LENGTH">
                <span>{{ COURSE_ID_MAX_LENGTH - newCourseId.length }} characters left</span>
            </div>
        </div>
    </div>
</div>
<div class="modal-footer">
    <button type="button" class="btn btn-light" (click)="activeModal.dismiss()">Cancel</button>
    <button id="btn-confirm-copy-course" type="button" class="btn btn-primary" (click)="copy()"
            [disabled]="!newCourseId">Copy</button>
</div>
