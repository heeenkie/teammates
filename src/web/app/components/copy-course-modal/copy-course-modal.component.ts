import {Component, OnInit} from '@angular/core';
import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
import { COURSE_ID_MAX_LENGTH } from '../../../types/field-validator';

/**
 * Copy current course modal.
 */
@Component({
  selector: 'tm-copy-course-modal',
  templateUrl: './copy-course-modal.component.html',
  styleUrls: ['./copy-course-modal.component.scss']
})
export class CopyCourseModalComponent implements OnInit {

  // const
  public COURSE_ID_MAX_LENGTH: number = COURSE_ID_MAX_LENGTH;

  public newCourseId : string = '';

  constructor(public activeModal: NgbActiveModal) { }

  ngOnInit(): void {
  }

  /**
   * Fires the copy event.
   */
  copy(): void {
    this.activeModal.close({
      newCourseId: this.newCourseId,
    });
  }

}
