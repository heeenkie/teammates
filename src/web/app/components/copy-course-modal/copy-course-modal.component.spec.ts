import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';
import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
import { CopyCourseModalComponent } from './copy-course-modal.component';
import { By } from '@angular/platform-browser';


describe('CopyCourseModalComponent', () => {
  let component: CopyCourseModalComponent;
  let fixture: ComponentFixture<CopyCourseModalComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ CopyCourseModalComponent ],
      imports: [
        FormsModule,
      ],
      providers: [
        NgbActiveModal,
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CopyCourseModalComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with default fields', () => {
    expect(fixture).toMatchSnapshot();
  });



  it('should snap with some course id', () => {
    component.newCourseId = "Test02";
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();

  });

  it('should enable copy button after new courseId is provided', () => {
    component.newCourseId = "Test02";
    fixture.detectChanges();
    const copyButton: any = fixture.debugElement.query(By.css('button.btn.btn-primary'));
    expect(copyButton.nativeElement.disabled).toBeFalsy();
  });

  it('should disable copy if courseId is empty', () => {
    component.newCourseId = "";
    fixture.detectChanges();
    const copyButton: any = fixture.debugElement.query(By.css('button.btn.btn-primary'));
    expect(copyButton.nativeElement.disabled).toBeTruthy();
  });

});
