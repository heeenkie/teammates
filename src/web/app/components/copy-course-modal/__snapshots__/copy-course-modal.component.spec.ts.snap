// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyCourseModalComponent should snap with default fields 1`] = `
<tm-copy-course-modal
  COURSE_ID_MAX_LENGTH={[Function Number]}
  activeModal={[Function NgbActiveModal]}
  newCourseId=""
>
  <div
    class="modal-header"
  >
    <h5
      class="modal-title"
    >
      <div>
        Copy this course with sessions
      </div>
    </h5>
    <button
      class="close"
      type="button"
    >
      <i
        class="fas fa-times"
      />
    </button>
  </div><div
    class="modal-body"
    id="copy-course-modal"
  >
    <div
      class="row"
    >
      <div
        class="col-12"
      >
        <div
          class="form-group"
        >
          <label>
            <b>
              Id for copied course
            </b>
          </label>
          <input
            class="form-control ng-untouched ng-pristine ng-valid"
            id="copy-course-id"
            maxlength="40"
            type="text"
          />
          <span>
            40 characters left
          </span>
        </div>
      </div>
    </div>
  </div><div
    class="modal-footer"
  >
    <button
      class="btn btn-light"
      type="button"
    >
      Cancel
    </button>
    <button
      class="btn btn-primary"
      disabled=""
      id="btn-confirm-copy-course"
      type="button"
    >
      Copy
    </button>
  </div>
</tm-copy-course-modal>
`;

exports[`CopyCourseModalComponent should snap with some course id 1`] = `
<tm-copy-course-modal
  COURSE_ID_MAX_LENGTH={[Function Number]}
  activeModal={[Function NgbActiveModal]}
  newCourseId={[Function String]}
>
  <div
    class="modal-header"
  >
    <h5
      class="modal-title"
    >
      <div>
        Copy this course with sessions
      </div>
    </h5>
    <button
      class="close"
      type="button"
    >
      <i
        class="fas fa-times"
      />
    </button>
  </div><div
    class="modal-body"
    id="copy-course-modal"
  >
    <div
      class="row"
    >
      <div
        class="col-12"
      >
        <div
          class="form-group"
        >
          <label>
            <b>
              Id for copied course
            </b>
          </label>
          <input
            class="form-control ng-untouched ng-pristine ng-valid"
            id="copy-course-id"
            maxlength="40"
            type="text"
          />
          <span>
            34 characters left
          </span>
        </div>
      </div>
    </div>
  </div><div
    class="modal-footer"
  >
    <button
      class="btn btn-light"
      type="button"
    >
      Cancel
    </button>
    <button
      class="btn btn-primary"
      id="btn-confirm-copy-course"
      type="button"
    >
      Copy
    </button>
  </div>
</tm-copy-course-modal>
`;
